scripts/publish-release.sh

```diff
@@ -15,7 +15,7 @@ fi
 if [[ $(git describe --exact-match 2> /dev/null || :) =~ -canary ]];
 then
   echo "Publishing canary"
-  yarn run lerna publish from-git --npm-tag canary --yes
+  yarn run lerna publish from-git --npm-tag canary --no-git-reset --yes
```

Comment:
`--no-git-reset` makes lerna okay with `git update-index --skip-worktree`. It seems like we would be interested in the feature proposed in this [issue.](https://github.com/lerna/lerna/issues/2329)